pkg/workload/workload.go

```diff
@@ -756,7 +756,7 @@ func IsActive(w *kueue.Workload) bool {
 // IsEvictedByDeactivation returns true if the workload is evicted by deactivation.
 func IsEvictedByDeactivation(w *kueue.Workload) bool {
 	cond := apimeta.FindStatusCondition(w.Status.Conditions, kueue.WorkloadEvicted)
-	return cond != nil && cond.Status == metav1.ConditionTrue && cond.Reason == kueue.WorkloadEvictedByDeactivation
+	return cond != nil && cond.Status == metav1.ConditionTrue && strings.HasPrefix(cond.Reason, kueue.WorkloadEvictedByDeactivation)
```

Comment:
This is a bit surprising, but I see why needed. This is because of the concatenation we introduced here: https://github.com/kubernetes-sigs/kueue/commit/e5a73fbe3dca2ab59742acd47e2f712b682477f9#diff-60dd240c20adbd6a189d018d1c216c2d296730f446c341d8bf449fa6657964ffR193. So, I thought for a moment this was a bug, but luckily this function is only used in tests. cc @mbobrovskyi @PBundyra .